chain/chain/src/garbage_collection.rs

```diff
@@ -264,24 +265,28 @@ impl ChainStore {
                 }
                 debug_assert_eq!(blocks_current_height.len(), 1);
 
-                chain_store_update.clear_block_data(
-                    epoch_manager.as_ref(),
-                    *block_hash,
-                    GCMode::Canonical(tries.clone()),
-                )?;
-                gc_parent_shard_after_resharding(
-                    &mut chain_store_update,
+                let mut shards_to_cleanup =
+                    get_gc_state(&store, epoch_manager.as_ref(), block_hash, shard_tracker, me)?;
+                let parents_shards_to_cleanup = get_gc_parent_shard_after_resharding(
+                    &store,
                     epoch_manager.as_ref(),
                     block_hash,
                 )?;
-                gc_state(
-                    &mut chain_store_update,
+                shards_to_cleanup.extend(parents_shards_to_cleanup);
+
+                chain_store_update.clear_block_data(
```

Comment:
Move this to AFTER calling `get_gc_parent_shard_after_resharding` and `get_gc_state`

Reply:
also moving `clear_block_data` doesn't make any difference since it does not commit any changes, commit happens at the end of the loop 🙃

Reply:
Thanks! Reverted the changes as per discussion offline